temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowService.java

```diff
@@ -257,44 +259,51 @@ StartWorkflowExecutionResponse startWorkflowExecutionImpl(
     lock.lock();
     try {
       String newRunId = UUID.randomUUID().toString();
+      StartWorkflowExecutionResponse dedupedResponse = dedupedRequest(startRequest);
+      if (dedupedResponse != null) {
+        return dedupedResponse;
+      }
+
       existing = executionsByWorkflowId.get(workflowId);
       if (existing != null) {
         WorkflowExecutionStatus status = existing.getWorkflowExecutionStatus();
 
-        if (status == WorkflowExecutionStatus.WORKFLOW_EXECUTION_STATUS_RUNNING
-            && (reusePolicy == WorkflowIdReusePolicy.WORKFLOW_ID_REUSE_POLICY_TERMINATE_IF_RUNNING
-                || conflictPolicy
-                    == WorkflowIdConflictPolicy.WORKFLOW_ID_CONFLICT_POLICY_TERMINATE_EXISTING)) {
-          existing.terminateWorkflowExecution(
-              TerminateWorkflowExecutionRequest.newBuilder()
-                  .setNamespace(startRequest.getNamespace())
-                  .setWorkflowExecution(existing.getExecutionId().getExecution())
-                  .setReason("TerminateIfRunning WorkflowIdReusePolicy Policy")
-                  .setIdentity("history-service")
-                  .setDetails(
-                      Payloads.newBuilder()
-                          .addPayloads(
-                              Payload.newBuilder()
-                                  .setData(
-                                      ByteString.copyFromUtf8(
-                                          String.format("terminated by new runID: %s", newRunId)))
-                                  .build())
-                          .build())
-                  .build());
-        } else if (status == WorkflowExecutionStatus.WORKFLOW_EXECUTION_STATUS_RUNNING
-            && conflictPolicy
-                == WorkflowIdConflictPolicy.WORKFLOW_ID_CONFLICT_POLICY_USE_EXISTING) {
-          return StartWorkflowExecutionResponse.newBuilder()
-              .setStarted(false)
-              .setRunId(existing.getExecutionId().getExecution().getRunId())
-              .build();
-        } else if (status == WorkflowExecutionStatus.WORKFLOW_EXECUTION_STATUS_RUNNING
-            || reusePolicy == WorkflowIdReusePolicy.WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE) {
-          return throwDuplicatedWorkflow(startRequest, existing);
-        } else if (reusePolicy
-                == WorkflowIdReusePolicy.WORKFLOW_ID_REUSE_POLICY_ALLOW_DUPLICATE_FAILED_ONLY
-            && (status == WorkflowExecutionStatus.WORKFLOW_EXECUTION_STATUS_COMPLETED
-                || status == WorkflowExecutionStatus.WORKFLOW_EXECUTION_STATUS_CONTINUED_AS_NEW)) {
+        if (status == WORKFLOW_EXECUTION_STATUS_RUNNING) {
+          if (reusePolicy == WORKFLOW_ID_REUSE_POLICY_TERMINATE_IF_RUNNING
```

Comment:
There's some edge cases that we handled in the server (or at least planned to) for reuse policy terminate if running and the different conflict policies.

Someone can address that in a follow up PR though.

CC @rodrigozhou

Reply:
@rodrigozhou have you reviewed this code and verified the different edge cases?
Can wait for a follow up PR.

Reply:
I'll do in a follow up PR, this one is already quite big.